sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -77,6 +80,12 @@ private[hive] object SparkSQLCLIDriver extends Logging {
     })
   }
 
+  private def isSecuredAndProxy(hiveConf: HiveConf): Boolean = {
```

Comment:
Isn't this basically `HiveDelegationTokenProvider.delegationTokensRequired`? Doesn't it work if you just call that method? The only difference is the check for deploy mode, which should work fine in this context.